tarantool/dbapi.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+import re
+
+from tarantool.error import InterfaceError
+from .connection import Connection as BaseConnection
+
+
+update_insert_pattern = re.compile(r'^UPDATE|^INSERT', re.IGNORECASE)
+
+
+class Cursor:
+    _lastrowid = 0
+    _rowcount = 0
+    description = None
+    position = 0
+    arraysize = 1
+    autocommit = True
```

Comment:
The standard says:
> Database modules that do not support transactions should implement this method with void functionality.

if the commit method has void functionality, autocommit attribute does not make any sense, so I removed this attribute.

Reply:
So the standard requires one default behaviour for databases with transactions support and another one for databases without it. It becomes a problem if a database did not support transactions, but than implemented them: the default behaviour must be changed and it is incompatible change. I propose to do our best:

* Provide `autocommit` connection property with default value `True`. An attempt to set the property to `False` should raise NotSupportedError, but it may be set to `True`. Also it worth to raise an InterfaceError on attempt to set the property to a non-boolean value or `None`.
* Suggest to verify or explicitly set the property to `True` in code that rely on the autocommit mode behaviour.

I'll include the following paragraph to the Databse API wiki page (not created yet) and to future release notes:

> The module enables autocommit mode by default, which is reflected in the `<connection object>.autocommit` property. This default will be changed when [interactive transactions support](https://github.com/tarantool/tarantool/issues/2016) will land into tarantool and will implemented in the module (it is PEP 249 requirement). Code that rely on the autocommit mode should either verify the property or explicitly set it to `True`.

Reply:
According to your suggestions I implemented auto-commit property and corresponding setter:
```python
@property
def autocommit(self):
    """Autocommit is True by default because Tarantool
    does not support interactive transactions
    but it will be implemented soon."""
    return self._autocommit

@autocommit.setter
def autocommit(self, autocommit):
    """Set autocommit state"""
    if not isinstance(autocommit, bool):
        raise InterfaceError("autocommit parameter must be boolean, "
                             "not %s" % type(autocommit))
    if autocommit is False:
        raise NotSupportedError("Can not The connector supports "
                                "only autocommit mode")
```
What do you think?

Reply:
It follows my idea, yep. But, please, take glance at messages:

> but it will be implemented soon

IMHO, the key point should be '<...> and the default will be changed to False. Set the property explicitly to True or verify it when lean on autocommit behaviour'. Feel free to reword, it is just for the idea.

> Can not The connector supports only autocommit mode

Looks as a typo.

BTW, I published my current draft of the Database API status to wiki: https://github.com/tarantool/tarantool-python/wiki/PEP-249-Database-API

It may contain mistakes, need to proof-read against current implementation. But it is surely better than nothing, so I decided to publish it now.

Reply:
FIxed the error messages, thanks.

Wiki looks nice, I'll add some notes there.